### ISSUE TITLE
markdown table not rendered

### ISSUE THREAD

**Issue by amueller** [nteract/archived-desktop-app#1557]
```
| Age | Number of Cars owned | Owns House | Number of Children| Marital status (s/m/d/w) | Owns a dog | Bought a Boat|
|-|-|-|-|-|-|-|
|66|1|y|2|w|n|y|
|52|2|y|3|m|n|y|
|22|0|n|0|m|y|n|
|25|1|n|1|s|n|n|
|44|0|n|2|d|y|n|
|39|1|y|2|m|y|n|
|26|1|n|2|s|n|n|
|40|3|y|1|m|y|n|
|53|2|y|2|d|n|y|
|64|2|y|3|d|n|n|
|58|2|y|2|m|y|y|
|33|1|n|1|s|n|n|
```
![image](https://cloud.githubusercontent.com/assets/449558/24009419/52b3d2de-0a4b-11e7-92b1-baa0e11196fc.png)

**Comment by rgbkrk** [comment]
Now that [GitHub released a formal spec for GitHub flavored Markdown](https://githubengineering.com/a-formal-spec-for-github-markdown/) we'll likely adopt it, which would include tables. There's not a JavaScript implementation that is compliant yet (this one is written in C), I'm looking forward to it though!

**Comment by amueller** [comment]
What does Jupyter Notebook use? I think aiming at being as compatible as reasonably possible would be good. This renders fine in Jupyter.

**Comment by amueller** [comment]
I know in the backend jupyter uses pandoc but not for browser stuff, right?

**Comment by amueller** [comment]
somewhat off-topic: I would like my notebooks to have a layout language that allows me to refer to the input or output of a cell and to sections in the text. I want a cell to be able to say "see In[10]" in a dynamic way, or refer to sections. Markdown doesn't allow that and I think GFM doesn't allow that either.
It's a pretty big limitation imho :-/
[Oh and I don't want to hack this with html because I want my latex conversions to also work and not just drop it]

**Comment by rgbkrk** [comment]
> I know in the backend jupyter uses pandoc but not for browser stuff, right?

That's mostly correct, depends on what's being used in nbconvert.

> What does Jupyter Notebook use?

Jupyter notebook uses [`marked`](https://github.com/chjj/marked), which is not as maintained as we would like it to be. It also has subtle differences with the spec for commonmark (ex-standard markdown). We're discussing the future in Jupyter there as well and want to be as close to the formal specification for commonmark as well.

**Comment by rgbkrk** [comment]
> I would like my notebooks to have a layout language that allows me to refer to the input or output of a cell and to sections in the text

Me too! We'll need a totally separate notebook format for that though. No time for me personally to charter or implement that.

**Comment by amueller** [comment]
Ok. My point is mostly: I like for stuff to look the same in nteract as it does in Jupyter Notebooks. I'm not sure what the timeline is for the GFM implementation in JavaScript but I think it would be good to prioritize this to make transition from Jupyter Notebooks as easy as possible.

**Comment by ProgramFan** [comment]
How about integrating commonmark-table-extension https://github.com/webuni/commonmark-table-extension?

**Comment by jdetle** [comment]
^ It seems this is for PHP? I don't know how we'd port that extension to javascript.

**Comment by ProgramFan** [comment]
Sorry to have made the mistake. For pure JS implementation, [Remarkable](https://github.com/jonschlinkert/remarkable) looks promissing. It support full GFM. Tables are handled. It also brings syntax highlighting for code blocks.

**Comment by jdetle** [comment]
@ProgramFan are you a student at an accredited university? A proposal for a Google Summer of Code project to make nteract conform with the jupyter notebook would be great.

**Comment by ProgramFan** [comment]
Unfortunately not any more 😃 @jdetle. I work full time as an associate professor and jupyter helps me a lot with my research, and I find nteract such a lovely tool. I like the proposal but just have not yet found the time. Maybe I can persuade some one to try it.

**Comment by jdetle** [comment]
If you find a graduate or undergraduate student who is interested in programming please encourage them to apply as soon as possible! Thanks!

Link:  https://github.com/nteract/nteract/wiki/Google-Summer-of-Code

**Comment by ProgramFan** [comment]
Of course.

**Comment by SamRodkey** [comment]
I'm having the same issue.

**Comment by rgbkrk** [comment]
Closed by #2350.
